apps/OboeTester/app/src/main/java/com/mobileer/oboetester/TestAudioActivity.java

```diff
@@ -820,43 +941,93 @@ protected void toastPauseError(int result) {
     }
 
     public void pauseAudio() {
-        int result = pauseNative();
+        synchronized (mAudioStateLock) {
+            pauseAudioLocked();
+        }
+    }
+
+    @GuardedBy("mAudioStateLock")
+    private void pauseAudioLocked() {
+        int result;
+        if (mAudioState != AUDIO_STATE_STARTED) {
+            Log.w(TAG, "pauseAudio() called in invalid state " + mAudioState + ", ignoring.");
+            return;
+        }
+        result = pauseNative();
+        if (result == 0) {
+            mAudioState = AUDIO_STATE_PAUSED;
+        }
+
         if (result != 0) {
             toastPauseError(result);
         } else {
-            mAudioState = AUDIO_STATE_PAUSED;
             updateEnabledWidgets();
             onStopAllContexts();
         }
     }
 
     public void flushAudio() {
-        int result = flushNative();
+        int result;
+        synchronized (mAudioStateLock) {
+            if (mAudioState != AUDIO_STATE_PAUSED) {
```

Comment:
I don't this we should check the state here and this is wrong. See https://cs.android.com/android/platform/superproject/main/+/main:frameworks/av/media/libaaudio/include/aaudio/AAudio.h;l=1874?q=aaudio.h

OboeTester should be able to call flush and let the lower layer return failure.